task7/index.js

```diff
@@ -0,0 +1,388 @@
+                }
+                else if (item === 'vendor') {
+                    elem = elem.filter(item => item.vendor === filterConfig.vendor);
+                }
+                else {
+                    return 'Error'
+                }
+            }
+        }
+        elem.sort(function (a, b) {
+            return a.createdAt - b.createdAt;
+        });
+        return elem.slice(skip, skip + top);
+    }
+
+
+    get(id) {
+        if (id.valueOf() > 0) {
+            return adList.find(item => item.id == id)
+        }
+        return false;
+    }
+
+    static _validate(elem) {
+        return typeof elem.id == 'string' && elem.id !== ""
+            && typeof elem.description == 'string' && elem.description.length <= 200 && typeof elem.link == 'string' && elem.description !== ""
+            && elem.createdAt instanceof Date && elem.createdAt !== ""
+            && typeof elem.link == 'string'
+            && typeof elem.vendor == 'string' && elem.vendor !== ""
+            && elem.validUntil instanceof Date
+            && Array.isArray(elem.hashTags)
+            && typeof elem.discount == 'string' && elem.discount !== ""
+            && elem.validUntil instanceof Date && elem.validUntil !== ""
+    }
+
+    add(newElem) {
+        if (adCollection._validate(newElem)) {
+            adList.push(newElem)
+            return adList
+        }
+        else return false
+    }
+
+
+    edit(id, propertyChange) {
+         for (let item in propertyChange) {
+            if (item === 'id' || item === 'vendor' || item === 'createdAt') {
+                return 'Error'
+            }
+            if (item === 'discount') {
+                adList.find(item => item.id === id).discount = propertyChange.discount
+                return true
+            }
+            if (item === 'link'){
+                adList.find(item => item.id === id).link = propertyChange.link
+                return true
+            }
+            if (item === 'validUntil'){
+                adList.find(item => item.id === id).validUntil = propertyChange.validUntil
+                return true
+            }
+            if (item === 'description'){
+                adList.find(item => item.id === id).description = propertyChange.description
+                return true
+            }
+            if (item === 'hashTags'){
+                adList.find(item => item.id === id).hashTags = propertyChange.hashTags
+                return true
+            }
+        }
+        return false
+    }
+
+    remove(id) {
+        if (typeof id === 'string') {
+            let i = adList.find(item => item.id === id);
+            adList.splice(i, 1)
+            return adList
+
+        } else return false
+    }
+
+    addAll(array){
+        let result = []
+        for(let item of this._adList) {
+            if (adCollection._validate(item)) {
+                array.push(item)
```

Comment:
I think it should be vice versa:  for each item of `array` you push it to `this._adList` in case item is valid, otherwise to `result`